lib/src/auth/mod.rs

```diff
@@ -0,0 +1,12 @@
+mod static_client_certificate_provider;
+
+#[derive(Debug, Clone)]
+pub struct ClientCertificate {
+    pub(crate) cert_file: String,  // Path to the TLS certificate file.
```

Comment:
If this is a path, then use `PathBuf` over `String` (which, IMHO, would also obviate the comment).